### PR TITLE
Change Root to non-unique impl pointer

### PR DESCRIPTION
**Pull request by nkoenig**
Signed-off-by: Nate Koenig <nate@openrobotics.org>

# 🎉 New feature

## Summary

Change Root from a unique impl pointer to a non-unique impl pointer. This will allow the Root object to be copyable.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [x] Updated documentation (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: include/sdf/Root.hh [hunk 1]
```diff
@@ -152,7 +152,7 @@ namespace sdf
     public: sdf::ElementPtr Element() const;
 
     /// \brief Private data pointer
-    IGN_UTILS_UNIQUE_IMPL_PTR(dataPtr)
+    IGN_UTILS_IMPL_PTR(dataPtr)
```
Comment (azeey) [inline include/sdf/Root.hh:155]: We'll need to define the copy ctor and copy assignment operator to properly handle frame graphs.

**Review by azeey — CHANGES_REQUESTED** [review]

File: include/sdf/Root.hh (hunk 1 above)
Comment (scpeters) [reply include/sdf/Root.hh]: to verify that frame graphs work, we would need a test that loads an `sdf::Root` from an SDFormat string or file and then copies it and tries to resolve a pose or attached-to frame to confirm that the graphs are properly copied

Comment (nkoenig) [reply include/sdf/Root.hh]: I think I'm missing something here. I added tests to capture frame copying, but the tests just passed without requiring the definition of a copy constructor and copy assignment operator. See [6b64370](https://github.com/gazebosim/sdformat/pull/844/commits/6b6437074cd8522543cc20152dc914e0743d1ce9).

File: src/Frame.cc [hunk 2]
```diff
@@ -217,3 +218,18 @@ sdf::ElementPtr Frame::Element() const
...
+{
+  sdf::ElementPtr elem(new sdf::Element);
+  sdf::initFile("frame.sdf", elem);
+
+  elem->GetAttribute("name")->Set(this->dataPtr->name);
+
+  if (!this->dataPtr->attachedTo.empty())
+    elem->GetAttribute("attached_to")->Set(this->dataPtr->attachedTo);
+
+  elem->GetElement("pose")->Set(this->dataPtr->pose);
```
Comment (scpeters) [inline src/Frame.cc:233]: I recommend copying the code from `Link::ToElement` that handles the relative_to attribute:

https://github.com/gazebosim/sdformat/blob/main/src/Link.cc#L655-L662

File: include/sdf/Root.hh (hunk 1 above)
Comment (scpeters) [reply include/sdf/Root.hh]: you're right; I commented this before I saw your latest commit; this is fine

File: src/Frame.cc (hunk 2 above)
Comment (nkoenig) [reply src/Frame.cc]: Done in 76d3794

**Review by scpeters — APPROVED** [review]

**Pull request merged**